overlord/snapstate/storehelpers.go

```diff
@@ -648,33 +657,37 @@ func storeUpdatePlanCore(
 			return updatePlan{}, err
 		}
 
-		compsups, err := componentSetupsForInstall(ctx, st, compsToInstall, *snapst, si.Revision, revOpts.Channel, opts)
+		// add the additional components that the caller requested to be
+		// installed
+		compsToInstall = unique(append(compsToInstall, up.AdditionalComponents...))
```

Comment:
do we have a test for this case of AdditionalComponents?

Reply:
Added a test for this case.